src/AzureAuth/CommandMain.cs

```diff
@@ -167,6 +173,31 @@ public Alias TokenFetcherOptions
             get { return this.authSettings; }
         }
 
+        /// <summary>
+        /// Gets the cache file name by given parameters or environment variables.
+        /// </summary>
+        public string WrappedCacheFilePath
+        {
+            get
+            {
+                // Check command parameter first.
+                if (!string.IsNullOrEmpty(this.CacheFilePath))
+                {
+                    return this.CacheFilePath;
+                }
+
+                // Check environment variable.
+                string envCacheFile = this.env.Get(EnvVars.AZUREAUTH_CACHE_FILE);
+                if (!string.IsNullOrEmpty(envCacheFile))
+                {
+                    return envCacheFile;
+                }
+
+                // Use default cache file name.
+                return $"msal_{this.authSettings.Tenant}.cache";
+            }
+        }
+

```

Comment:
I dislike that we need a separate function called `Wrapped***` here. Another option would be to implement this `get` function as the get method for the property itself (`CacheFilePath`) and use a private member to back the property. You have to implement `set` as well so that if the CLI parses the option, it can set it. But otherwise the private member will remain null, and you can memoize the env var during the `get` method. Does that make sense?